packages/coding-standard/src/Rules/NoFactoryInConstructorRule.php

```diff
@@ -52,12 +52,12 @@ public function process(Node $node, Scope $scope): array
 
             $expression = $stmt->expr;
             while ($expression) {
-                /** @var MethodCall $expression */
-                if ($expression instanceof MethodCall) {
+                /** @phpstan-ignore-next-line */
+                if ($expression instanceof MethodCall && ! in_array($expression->var->name, ['this', 'self', 'static'], true)) {
                     return [self::ERROR_MESSAGE];
                 }
 
-                /** @var MethodCall|null $expression */
+                /** @phpstan-ignore-next-line */
                 $expression = $expression->expr;
             }
         }
```

Comment:
What about using `NodeFinder` class to find `MethodCall`?

Reply:
I will try

Reply:
implemented with `NodeFinder`

Reply:
Very nice :+1: